libraries/chain/include/eosio/chain/apply_context.hpp

```diff
@@ -621,6 +627,11 @@ class apply_context {
       bool                          privileged   = false;
       bool                          context_free = false;
 
+      std::optional<sync_call_context> sync_call_ctx{};  // only one of act and sync_call_ctx can be present
+
+      // Returns the sender of any sync call initiated by this apply_context or its sync_call_ctx
+      action_name get_sync_call_sender() const;
```

Comment:
It is strange that this is `action_name` for a name that isn't an action. But I see it's the same for `get_sender()`

Reply:
In https://github.com/AntelopeIO/spring/pull/1273, `get_sync_call_sender()` is replaced with the virtual function `get_sender()`.